### PR TITLE
SAK-33098 add bulk edit for gradebook items

### PR DESCRIPTION
**Pull request by steveswinsburg**
Ref https://jira.sakaiproject.org/browse/SAK-33098

Screenshot:
screen shot 2018-10-08 at 10 39 23 pm

### CODE REVIEW

**Comment by ern** [comment]
@steveswinsburg can you add the changes requested by accesslint and lets keep the whitespace changes to a minimum while that conversation happens in the email list?

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/DeleteItemPanel.html [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
 			<p class="alert alert-warning" role="alert"><wicket:message key="delete.warning"></wicket:message></p>
 
-			<div class="">
-				<input type="submit" wicket:id="submit" wicket:message="value:button.deleteitem" />
+			<div class="act">
+				<input type="submit" wicket:id="submit" class="active" wicket:message="value:button.deleteitem" />
```
[the remark this answers is not in the document]
Comment (steveswinsburg) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/DeleteItemPanel.html]: false positive -> wicket:message="value:... sets the value attribute in the rendered output.

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+						<td><span wicket:id="itemTitle">item title</span></td>
+						<td><input wicket:id="release" type="checkbox" /></td>
+						<td><input wicket:id="include" type="checkbox" /></td>
+					</tr>
+				</tbody>
+			</table>
+			
+				
+			<div class="act">
+				<input type="submit" wicket:id="submit" class="active" wicket:message="value:button.savechanges" />
```
[the remark this answers is not in the document]
Comment (steveswinsburg) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html]: false positive -> wicket:message="value:... sets the value attribute in the rendered output.

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+					<tr>
+						<th class="col-md-4"><wicket:message key="bulkedit.column.header.gradebookitem" /></th>
+						<th class="col-md-1"><wicket:message key="bulkedit.column.header.release" /></th>
+						<th class="col-md-1"><wicket:message key="bulkedit.column.header.include" /></th>
+					</tr>
+				</thead>
+				<tbody>
+					<tr wicket:id="listView">
+						<td><span wicket:id="itemTitle">item title</span></td>
+						<td><input wicket:id="release" type="checkbox" /></td>
```
[the remark this answers is not in the document]
Comment (steveswinsburg) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html]: Fixed

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+						<td><input wicket:id="include" type="checkbox" /></td>
```
[the remark this answers is not in the document]
Comment (steveswinsburg) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html]: Fixed

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+					</tr>
+				</tbody>
+			</table>
+			
+				
+			<div class="act">
+				<input type="submit" wicket:id="submit" class="active" wicket:message="value:button.savechanges" />
+				<input type="submit" wicket:id="cancel" wicket:message="value:button.cancel" />
```
[the remark this answers is not in the document]
Comment (steveswinsburg) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/BulkEditItemsPanel.html]: False positive. wicket:message="value:... sets the value for this.

File: edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/Assignment.java [hunk 6]
```diff
@@ -24,87 +24,143 @@
...
 @NoArgsConstructor
-@ToString
 public class Assignment implements Serializable, Comparable<Assignment> {
 	private static final long serialVersionUID = 1L;
 
 	/**
 	 * the id of the assignment in the gradebook
 	 */
-    @Getter @Setter private Long id;
+	@Getter
```
Comment (ern) [inline edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/Assignment.java]: Hi Steve this file has been reformatted using Tabs, spaces are preferred...

**Comment by ern** [comment]
This was discussed in the Core Meeting
see https://etherpad.nightly.sakaiproject.org/p/core-2018-10-16
and a consensus was reached that it doesn't make sense to accept a PR with large amounts of non functional changes.

**Comment by steveswinsburg** [comment]
I am in the middle of updating. Please do not close my PRs without consulting me first.

**Comment by steveswinsburg** [comment]
Codacy issues unrelated to this change.

**Pull request merged**